analytics/lib/counts.py

```diff
@@ -38,6 +40,16 @@
 # You can't subtract timedelta.max from a datetime, so use this instead
 TIMEDELTA_MAX = timedelta(days=365 * 1000)
 
+
+def end_of_month(dt: datetime) -> datetime:
+    year = dt.year
+    month = dt.month
+    _, last_day = calendar.monthrange(year, month)
+    return datetime(
+        year, month, last_day, hour=23, minute=59, second=59, microsecond=999, tzinfo=timezone.utc
```

Comment:
I can add tests for this, if this logic looks good to you.

Reply:
See below, I don't think we need this function.